### PR TITLE
Refactor client to use multiple inheritance

### LINKED ISSUE #33: Explore different ways to dynamically populate the Client class
In #31, we introduced some metaprogramming to split the Client class into modules by endpoint. As raised in #32, using inspect may not be the best way to do this.

### PR DESCRIPTION
**Pull request by dinahshi**
Attempt to fix #33. Uses the idea of a LowLevelClient as @mittonk suggested in the thread.

Structure based on [octokit](https://github.com/octokit/octokit.py/tree/master/octokit). The endpoints are still modular but the Client class inherits from them rather than dynamically copying methods. Note that helper methods are now accessible.

cc @asottile @tomelm

### CODE REVIEW

**Comment by tomelm** [comment]
Do the docs need to change here at all?

**Comment by dinahshi** [comment]
This shouldn't affect the previously exposed methods so I would say no.

File: yelp/client.py [hunk 1]
```diff
@@ -11,35 +10,13 @@
...
-    def _add_instance_methods(self, instance_methods):
-        # instance_methods is a list of (name, value) tuples where value is the
-        # instance of the bound method
-        for method in instance_methods:
-            if method[0][0] is not '_':
-                self.__setattr__(method[0], method[1])
-
-    def _make_request(self, path, url_params={}):
+
+    def make_request(self, path, url_params={}):
```
Comment (dinahshi) [inline yelp/client.py]: Perhaps this should be left as `_make_request` to discourage people from using it directly.

Comment (watterso) [inline yelp/client.py]: I agree, the people that should be touching this function directly are those contributing to the clientlib itself *not* the random developer on the internet that wants to integrate our data.

File: tests/endpoint/business_test.py [hunk 2]
```diff
@@ -13,14 +13,14 @@ def setup_class(cls):
         cls.client = Client(auth)
 
     def test_get_business_builds_correct_params(self):
-        with mock.patch('yelp.client.Client._make_request') as request:
+        with mock.patch('yelp.client.BaseClient.make_request') as request:
```
Comment (watterso) [inline tests/endpoint/business_test.py]: Non-blocking: this mock is used in quite a few places, consider making a decorator or something similar to DRY it out.

**Pull request merged**